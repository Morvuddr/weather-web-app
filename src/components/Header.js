import React from 'react';
import * as PropTypes from 'prop-types';

const Header = ({ onUpdate = f => f }) => (
    <header className='App-header'>
        <h1 className='App-title'>Погода здесь</h1>
        <button className='Refresh-Location-Button' onClick={onUpdate}>Обновить геолокацию</button>
    </header>
)

Header.propTypes = {
    onUpdate: PropTypes.func
}

export default Header;