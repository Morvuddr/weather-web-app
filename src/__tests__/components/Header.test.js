import Header from '../../components/Header';
import renderer from 'react-test-renderer';
import React from 'react';

describe('Header component', () => {
    test('matches the snapshot', () => {
        const tree = renderer.create(
            <Header />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});