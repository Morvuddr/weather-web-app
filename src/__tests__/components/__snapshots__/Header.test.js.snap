// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches the snapshot 1`] = `
<header
  className="App-header"
>
  <h1
    className="App-title"
  >
    Погода здесь
  </h1>
  <button
    className="Refresh-Location-Button"
    onClick={[Function]}
  >
    Обновить геолокацию
  </button>
</header>
`;
